homeassistant/components/netatmo/climate.py

```diff
@@ -114,27 +114,41 @@ def __init__(self, data, room_id):
         self._data = data
         self._state = None
         self._room_id = room_id
-        room_name = self._data.homedata.rooms[self._data.home][room_id]['name']
-        self._name = 'netatmo_{}'.format(room_name)
+        self._room_name = self._data.homedata.rooms[
+            self._data.home][room_id]['name']
+        self._name = 'netatmo_{}'.format(self._room_name)
         self._target_temperature = None
         self._away = None
-        self._module_type = self._data.room_status[room_id]['module_type']
-        if self._module_type == NA_VALVE:
-            self._operation_list = [DICT_NETATMO_TO_HA[STATE_NETATMO_SCHEDULE],
-                                    DICT_NETATMO_TO_HA[STATE_NETATMO_MANUAL],
-                                    DICT_NETATMO_TO_HA[STATE_NETATMO_AWAY],
-                                    DICT_NETATMO_TO_HA[STATE_NETATMO_HG]]
-            self._support_flags = SUPPORT_FLAGS
-        elif self._module_type == NA_THERM:
-            self._operation_list = [DICT_NETATMO_TO_HA[STATE_NETATMO_SCHEDULE],
-                                    DICT_NETATMO_TO_HA[STATE_NETATMO_MANUAL],
-                                    DICT_NETATMO_TO_HA[STATE_NETATMO_AWAY],
-                                    DICT_NETATMO_TO_HA[STATE_NETATMO_HG],
-                                    DICT_NETATMO_TO_HA[STATE_NETATMO_MAX],
-                                    DICT_NETATMO_TO_HA[STATE_NETATMO_OFF]]
-            self._support_flags = SUPPORT_FLAGS | SUPPORT_ON_OFF
         self._operation_mode = None
         self.update_without_throttle = False
+        self._module_type = None
+        self._support_flags = SUPPORT_FLAGS
+        self._operation_list = [
+            DICT_NETATMO_TO_HA[STATE_NETATMO_SCHEDULE],
+            DICT_NETATMO_TO_HA[STATE_NETATMO_MANUAL],
+            DICT_NETATMO_TO_HA[STATE_NETATMO_AWAY],
+            DICT_NETATMO_TO_HA[STATE_NETATMO_HG],
+        ]
+
+        _LOGGER.debug(
+            "Setting up thermostat %s in %s (%s).",
+            self._name, self._room_name, self._room_id
+        )
+        try:
+            self._module_type = self._data.room_status[room_id]['module_type']
+        except KeyError:
+            _LOGGER.error("Thermostat in %s not available.", room_id)
```

Comment:
You're not stopping initialization here, it means that operation list and support flags will never get initialized, resulting in a broken thermostat until reboot.

Reply:
We don't end logging messages with period.

Reply:
Operation list and support flags get initialised in 125-131, but you are right they would not be complete in case it is a thermostat and not a valve, which results in a subset of features.
But I agree, this is not ideal.